File: FurnitureAPI/FurnitureAPI/Models/FurnitureWithReviews.cs
using System.Collections.Generic;

namespace FurnitureAPI.Models
{
    public class FurnitureWithReviews
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public float Price { get; set; }

        public int FurnitureTypeId { get; set; }

        public string FurnitureTypeName { get; set; }

        public string Description { get; set; }

        public string Size { get; set; }

        public bool IsDeleted { get; set; }
        public string ImageUrl { get; set; }
        public int AvailableQuantity { get; set; }
        public float RatingAverage { get; set; }
        public int NumberOfReviews { get; set; }
        public int DiscountPercentage { get; set; }
        public int OrdersCount { get; set; }
        public int ReviewsPosition { get; set; }
        public int OrdersPosition { get; set; }


        public List<Review> Reviews { get; set; }

    }
}

File: FurnitureAPP/app/src/main/java/com/example/furnitureapp/models/FurnitureWithReviews.kt
package com.example.furnitureapp.models

import com.example.furnitureapp.datalayers.retrofit.models.ReviewModel
import com.google.gson.annotations.SerializedName

data class FurnitureWithReviews(
    val id: Int,
    val name: String,
    val imageUrl: String,
    val price: Float,
    val furnitureTypeId: Int,
    val furnitureTypeName: String,
    val description: String,
    val size: String,
    val availableQuantity: Int,
    val ratingAverage: Float,
    val numberOfReviews: Int,
    val discountPercentage: Int,
    val ordersCount: Int,
    val isDeleted: Boolean,
    val reviews: List<Review>,
    val reviewsPosition: Int,
    val ordersPosition: Int,
    val glbPath: String?,
    val glbScale: Float
)

